torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -306,10 +306,8 @@ def param_hook(
         if clean_key.startswith(clean_prefix):
             clean_key = clean_key[len(clean_prefix) :]
 
-        # Clone non-ignored parameters before exiting the `_unshard_params()` context.
-        if clean_key not in fsdp_state._ignored_param_names and not getattr(
-            state_dict[fqn], "_has_been_cloned", False
-        ):
+        # Clone parameters before exiting the `_unshard_params()` context.
```

Comment:
I wonder just for safety if we should have an assert ensuring that we're not processing an ignored parameters, but this would then again imply that we have to keep the names. Not sure if it's worth it.

Reply:
'_ignored_param_names' here is legacy before state_dict refactoring, I think it should be safe to remove it and it can simplify codes a little bit